### PR TITLE
Create 2023-01-17-mongodb-monitoring.md

### PR DESCRIPTION
**Pull request by shyamvalsan**
MongoDB monitoring use-case blog

### CODE REVIEW

File: blog/2023-01-17-mongodb-monitoring.md [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+### Connections
+ - **Connections Count**: The total number of connections to the MongoDB server. Current connections and available connections are displayed as separate dimensions. 
+
+![image](https://user-images.githubusercontent.com/24860547/212876099-6ceed111-9663-4cb6-89db-76d3c33bcf25.png)
+
+ - **Connections by Rate**: The rate of connections created per second to the MongoDB server. 
+
+This metric provides insight into the traffic that is coming in to the MongoDB server. This will help identify spikes in traffic that could be caused by malicious activity or a sudden surge in usage from legitimate users. By monitoring this metric, you can identify potential bottlenecks or issues before they become problematic. Additionally, it can help you identify when you need to scale your MongoDB server to accommodate additional traffic. By monitoring this metric, you can prevent slowdowns and other issues related to an overloaded server.
+
+ - **Connections by State**: The distribution of connection states for the MongoDB server. The following states are displayed as separate dimensions: active, threaded, exhaustIsMaster, exhaustHello, awaiting_topology_changes.
```
Comment (sashwathn) [inline blog/2023-01-17-mongodb-monitoring.md:56]: Chart snapshot missing?

File: blog/2023-01-17-mongodb-monitoring.md [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+High latencies may be caused by a number of different factors, such as inadequate hardware resources, inefficient queries, or contention between concurrent operations. By monitoring this metric, it is possible to quickly identify any potential performance issues and take appropriate corrective action. Monitoring this metric can help to avoid any unexpected downtime or performance degradation.
+
+![image](https://user-images.githubusercontent.com/24860547/212876030-46a49c5a-8d7b-4cc2-b040-60638cdd043a.png)
+
+### Connections
+ - **Connections Count**: The total number of connections to the MongoDB server. Current connections and available connections are displayed as separate dimensions. 
+
+![image](https://user-images.githubusercontent.com/24860547/212876099-6ceed111-9663-4cb6-89db-76d3c33bcf25.png)
+
+ - **Connections by Rate**: The rate of connections created per second to the MongoDB server.
```
Comment (sashwathn) [inline blog/2023-01-17-mongodb-monitoring.md:52]: Chart snapshot missing?

File: blog/2023-01-17-mongodb-monitoring.md [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+ - **Network Requests**: The number of requests per second to the MongoDB server. 
+ 
+If the number of requests is too high, it can cause the database to become overloaded and degrade performance. Conversely, if the number of requests is too low, it could indicate that the server isn't being utilized efficiently. By monitoring the number of requests per second, you can ensure that the server is operating at an optimal level and identify any discrepancies in usage patterns. 
+
+For example, if the number of requests suddenly increases, it could be indicative of a problem with the database or a spike in usage. If the number of requests drops suddenly, it could mean that the database is not being utilized or that the application is experiencing an issue. By monitoring network requests, you can quickly identify any issues and take the necessary steps to rectify them.
+ 
+![image](https://user-images.githubusercontent.com/24860547/212876243-f96fae2a-b5c1-4ecc-b425-8a9dbd38424b.png)
+
+### Memory
+ - **Page Faults**: The number of page faults encountered by the MongoDB server per second.
```
Comment (sashwathn) [inline blog/2023-01-17-mongodb-monitoring.md:76]: Chart snapshot missing?

File: blog/2023-01-17-mongodb-monitoring.md [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+
+For example, if the pageheap_free metric is consistently low and pageheap_unmapped is consistently high, it could indicate that memory is not being released back to the operating system, resulting in memory fragmentation. This can cause performance issues and lead to out-of-memory errors.
+
+Monitoring the total_threaded_cache metric can help identify when more memory is needed, as it shows the total amount of memory allocated by TCMalloc in bytes. If this metric is consistently high, it could indicate that more memory is needed to handle the workload.
+
+Monitoring the free metric can also help indicate when more memory is needed, as it shows the amount of free memory available to TCMalloc in bytes. If this metric is consistently low, it could indicate that more memory is needed to handle the workload.
```
Comment (sashwathn) [inline blog/2023-01-17-mongodb-monitoring.md:101]: Is this really the description coming from the chart info? May be too long for a Netdata user who already knows the subject.

File: blog/2023-01-17-mongodb-monitoring.md [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+![image](https://user-images.githubusercontent.com/24860547/212876358-78cd4505-4188-4b78-a524-5bf8b697ebf6.png)
+
+ - **TCMalloc Generic**: The usage of the TCMalloc generic allocator by the MongoDB server. The size of the heap and currently allocated bytes are monitored.
+ 
+TCMalloc Generic is an allocator that is used by MongoDB to manage memory use within the server. It helps to improve the performance of MongoDB by allowing for more efficient memory management. By monitoring the size of the heap, as well as the currently allocated bytes, it is possible to identify memory leaks, fragmentation, and other issues that may be causing performance problems. By monitoring this metric, it is possible to identify these issues before they become too severe, and take corrective action to address them. This can help to ensure that MongoDB performance remains optimal, and that any potential issues are quickly and efficiently addressed.
+ 
+![image](https://user-images.githubusercontent.com/24860547/212876295-61b834aa-97e9-4b4d-9e60-4520a807a5e7.png)
+
+### Asserts
+ - **Asserts**: The number of asserts encountered by the MongoDB server per second. The following assert types are monitored:
```
Comment (sashwathn) [inline blog/2023-01-17-mongodb-monitoring.md:112]: Chart snapshot missing?

File: blog/2023-01-17-mongodb-monitoring.md [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+	- user
+	- tripwire
+	- rollovers
+
+Asserts are MongoDB's way of detecting errors within the system. They are essentially checks that the system performs to ensure that data is being handled properly. Asserts can be caused by things such as incorrect data formats, incorrect user input, or a bug in the code. By monitoring the number of asserts encountered, it is possible to identify potential problems before they cause a disruption to service.
+
+For example, if the number of regular asserts increases significantly over a short period of time, it could indicate a bug in the code, or a data format issue. If this is the case, further investigation should be done in order to identify the root cause of the problem and prevent it from causing any service disruptions.
+
+### Transactions
+ - **Current Transactions** The number of current transactions on the MongoDB server. The following transaction types are monitored:
```
Comment (sashwathn) [inline blog/2023-01-17-mongodb-monitoring.md:125]: Chart snapshot missing?

File: blog/2023-01-17-mongodb-monitoring.md [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+
+### Clients
+ - **Active Clients**: The number of active clients (readers and writers) connected to the MongoDB server. 
+ 
+Monitoring this metric can help to identify when it is time to scale up the server to meet the demands of the workload. It can also help to identify any potential issues that may arise due to an overload of clients. For example, if the number of active clients is too high, it can result in requests timing out and errors being returned. By monitoring this metric, these issues can be identified and addressed quickly.
+  
+![image](https://user-images.githubusercontent.com/24860547/212876432-d8038c56-1381-478b-9cf1-bb7c8ef78950.png)
+
+ - **Queued Clients**: The number of clients (readers and writers) currently queued because of a lock on the MongoDB server.
```
Comment (sashwathn) [inline blog/2023-01-17-mongodb-monitoring.md:140]: Chart snapshot missing?

File: blog/2023-01-17-mongodb-monitoring.md [hunk 8]
```diff
@@ -0,0 +1,297 @@
...
+  
+![image](https://user-images.githubusercontent.com/24860547/212876432-d8038c56-1381-478b-9cf1-bb7c8ef78950.png)
+
+ - **Queued Clients**: The number of clients (readers and writers) currently queued because of a lock on the MongoDB server. 
+
+### Locks
+ - **Acquired Locks**: The distribution of locks acquired and held by the MongoDB server. The following locks are monitored global_read, global_write, database_read, database_write, collection_read, collection_write.
+ ![image](https://user-images.githubusercontent.com/24860547/212876511-50384d07-98f0-4e37-8112-63634604903d.png)
+ 
+ - **Flow control events**: The timings of flow control events (acquiring and lagged) on the MongoDB server.
```
Comment (sashwathn) [inline blog/2023-01-17-mongodb-monitoring.md:146]: Chart snapshot missing. Do these metrics require special configuration?

File: blog/2023-01-17-mongodb-monitoring.md [hunk 9]
```diff
@@ -0,0 +1,297 @@
...
+![image](https://user-images.githubusercontent.com/24860547/212877086-61fffbf7-2254-45c8-bb07-8ad5bf2601d0.png)
+
+ - **WiredTiger Transactions**:The number of transactions currently open in the WiredTiger storage engine on the MongoDB server. 
+ 
+Issues that can be prevented through effective monitoring of this metric include identifying slow queries, detecting potential deadlocks, and identifying any transactions that are taking longer than expected to commit.
+
+![image](https://user-images.githubusercontent.com/24860547/212877136-bc34c31f-827a-4366-9a5c-8907b9a5de82.png)
+
+### Database 
+ - **Database Collections**: The number of collections in the specified database on the MongoDB server.
```
Comment (sashwathn) [inline blog/2023-01-17-mongodb-monitoring.md:215]: Chart snapshot missing.. Good to document if specific configuration is needed on MongoDB? Comment valid for this entire section and subsequent sections on replication, shards etc.

**Review by sashwathn — APPROVED** [review]

File: blog/2023-01-17-mongodb-monitoring.md (hunk 1 above)
Comment (shyamvalsan) [reply blog/2023-01-17-mongodb-monitoring.md:56]: Got my screenshots from Ralph's space - but it doesn't have replication and some other stuff enabled. Some of the charts like this one just showed empty so I didn't get them.

File: blog/2023-01-17-mongodb-monitoring.md (hunk 4 above)
Comment (shyamvalsan) [reply blog/2023-01-17-mongodb-monitoring.md:101]: No, I ran it through another round of GPT to make it more verbose. Definitely too long for putting inside the app info description, but fine for a blog I think.

File: blog/2023-01-17-mongodb-monitoring.md (hunk 6 above)
Comment (shyamvalsan) [reply blog/2023-01-17-mongodb-monitoring.md:125]: same comment as above

File: blog/2023-01-17-mongodb-monitoring.md (hunk 7 above)
Comment (shyamvalsan) [reply blog/2023-01-17-mongodb-monitoring.md:140]: same comment as above

File: blog/2023-01-17-mongodb-monitoring.md (hunk 8 above)
Comment (shyamvalsan) [reply blog/2023-01-17-mongodb-monitoring.md:146]: same comment as above - learn doc mentions no special configs, I think its just about what features are enabled in MongoDB.

File: blog/2023-01-17-mongodb-monitoring.md (hunk 2 above)
Comment (shyamvalsan) [reply blog/2023-01-17-mongodb-monitoring.md:52]: same comment as above

File: blog/2023-01-17-mongodb-monitoring.md (hunk 3 above)
Comment (shyamvalsan) [reply blog/2023-01-17-mongodb-monitoring.md:76]: same comment as above

**Pull request merged**